src/PostgREST/Query/QueryBuilder.hs

```diff
@@ -82,12 +82,13 @@ getSelectsJoins rr@(Node ReadPlan{select, relName, relToParent=Just rel, relAggA
   in
   (if null select && null forest then selects else sel:selects, joi:joins)
 
-mutatePlanToQuery :: MutatePlan -> SQL.Snippet
-mutatePlanToQuery (Insert mainQi iCols body onConflct putConditions returnings _ applyDefaults) =
+mutatePlanToQuery :: Bool -> MutatePlan -> SQL.Snippet
+mutatePlanToQuery mergeDups (Insert mainQi iCols body onConflct putConditions returnings _ applyDefaults) =
```

Comment:
Maybe I'm being too nitpicky, but we could check if the preference is applied without passing `mergeDups` as a parameter, by using `Insert{onConflict}`:

```haskell
where
  -- ...
  mergeDups = case onConflct of {Just (MergeDuplicates, _) -> True; _ -> False;}
```

It would make the `mutatePlanToQuery` function cleaner in my eyes, WDYT?

---

Those are the only reviews I have. After this, then this PR is ready to be merged.

Reply:
@laurenceisla No, you're not being nitpicky. It is actually a really nice refactor. Code cleanup should always be done, better now than later.